### PR TITLE
Task-45335: Display scheduled article

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**
Prior to this change, when we schedule a news, we redirect to the space stream page.  With this PR we will redirect to a dedicated scheduled news page.

### CODE REVIEW

File: services/src/main/resources/locale/portlet/news/News_en.properties [hunk 1]
```diff
@@ -31,6 +31,7 @@ news.composer.schedule=Schedule
 news.composer.scheduleSuccess=Post scheduled is displayed
 news.composer.btn.scheduleArticle=Scheduled article
 news.activity.scheduled=Scheduled
+news.composer.btn.backToHome=Back to Home
```
Comment (azayati) [inline services/src/main/resources/locale/portlet/news/News_en.properties]: not needed

File: webapp/src/main/webapp/news-details-app/components/ExoNewsDetailsApp.vue [hunk 2]
```diff
@@ -1,6 +1,12 @@
 <template>
   <div id="newsDetail">
+    <div v-if="notFound" class="articleNotFound">
+      <i class="iconNotFound"></i>
+      <h3>{{ $t('news.app.noNews') }}</h3>
+      <v-btn class="btn btn-primary" @click="redirectUrl">{{ $t("news.composer.btn.backToHome") }}</v-btn>
```
Comment (azayati) [inline webapp/src/main/webapp/news-details-app/components/ExoNewsDetailsApp.vue]: not neeeded

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 3]
```diff
@@ -1163,16 +1166,21 @@ public News scheduleNews(News news) throws Exception {
...
         scheduledNewsNode.save();
         publicationService.changeState(scheduledNewsNode, PublicationDefaultStates.STAGED, new HashMap<>());
       }
+      scheduledNews = convertNodeToNews(scheduledNewsNode, false);
     } finally {
       if (session != null) {
         session.logout();
       }
     }
+    if (news.isPinned()) {
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: not needed IMHO after removing the pin action from news editor

File: services/src/main/java/org/exoplatform/news/NewsServiceImpl.java [hunk 4]
```diff
@@ -1163,16 +1166,21 @@ public News scheduleNews(News news) throws Exception {
         SimpleDateFormat format = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss" + "Z");
         Calendar startPublishedDate = Calendar.getInstance();
         startPublishedDate.setTime(format.parse(schedulePostDate));
+        scheduledNewsNode.setProperty("exo:pinned", news.isPinned());
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/NewsServiceImpl.java]: not needed IMHO after removing the pin action from news editor

File: services/src/main/resources/locale/portlet/news/News_en.properties [hunk 5]
```diff
@@ -29,6 +29,9 @@ news.composer.postLater=Post later
 news.composer.choosePostDate=Choose when you want your article to be posted:
 news.composer.schedule=Schedule
 news.composer.scheduleSuccess=Post scheduled is displayed
+news.composer.btn.scheduleArticle=Scheduled article
+news.activity.scheduled=Scheduled
```
Comment (azayati) [inline services/src/main/resources/locale/portlet/news/News_en.properties]: put this bundle below with the block news.details
```suggestion
news.details.scheduled=Scheduled
```

File: webapp/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 6]
```diff
@@ -144,6 +144,38 @@
     </module>
   </portlet>
 
+  <portlet>
+    <name>NewsDetail</name>
```
Comment (azayati) [inline webapp/src/main/webapp/WEB-INF/gatein-resources.xml:148]:
```suggestion
    <name>NewsDetails</name>
```

File: webapp/src/main/webapp/WEB-INF/portlet.xml [hunk 7]
```diff
@@ -62,4 +62,22 @@
         </portlet-info>
     </portlet>
 
+    <portlet>
+        <portlet-name>NewsDetail</portlet-name>
```
Comment (azayati) [inline webapp/src/main/webapp/WEB-INF/portlet.xml:66]:
```suggestion
        <portlet-name>NewsDetails</portlet-name>
```

File: webapp/src/main/webapp/WEB-INF/conf/news/portal/portal/global/pages.xml [hunk 8]
```diff
@@ -78,50 +78,8 @@
...
-              <read-only>false</read-only>
-            </preference>
-            <preference>
-              <name>sharedCache</name>
-              <value>false</value>
-              <read-only>false</read-only>
-            </preference>
-          </preferences>
+          <application-ref>news</application-ref>
+          <portlet-ref>NewsDetail</portlet-ref>
```
Comment (azayati) [inline webapp/src/main/webapp/WEB-INF/conf/news/portal/portal/global/pages.xml:82]:
```suggestion
          <portlet-ref>NewsDetails</portlet-ref>
```

File: webapp/src/main/webapp/components/snackbar/ExoNewsNotificationAlerts.vue [hunk 9]
```diff
@@ -58,6 +58,15 @@ export default {
         });
       }
     });
+    this.$root.$on('restricted-space', (spaceDisplayName) => {
```
Comment (azayati) [inline webapp/src/main/webapp/components/snackbar/ExoNewsNotificationAlerts.vue:61]:
```suggestion
    this.$root.$on('restricted-news-activity-access', (spaceDisplayName) => {
```
the message to be displayed will not contain space name

File: webapp/src/main/webapp/news-activity-composer-app/components/ExoNewsActivityComposer.vue [hunk 10]
```diff
@@ -559,7 +542,7 @@ export default {
         const captionText = this.$t('news.broadcast.action');
         const confirmButton = this.$t('news.broadcast.btn.confirm');
         const cancelButton = this.$t('news.edit.cancel');
-        eXo.social.PopupConfirmation.confirm('createdPinnedNews', [{action: this.doPostNews, label: confirmButton}], captionText, confirmText, cancelButton);
+        eXo.social.PopupConfirmation.confirm('createdPinnedNews', [{action: this.doPostNews(schedulePostDate), label: confirmButton}], captionText, confirmText, cancelButton);
```
Comment (azayati) [inline webapp/src/main/webapp/news-activity-composer-app/components/ExoNewsActivityComposer.vue:545]: not needed IMHO after removing the pin action from news editor

File: webapp/src/main/webapp/news-details-app/components/ExoNewsDetailsApp.vue [hunk 11]
```diff
@@ -0,0 +1,46 @@
+<template>
+  <div id="newsDetail">
+    <div v-if="notFound" class="articleNotFound">
+      <i class="iconNotFound"></i>
+      <h3>{{ $t('news.app.noNews') }}</h3>
```
Comment (azayati) [inline webapp/src/main/webapp/news-details-app/components/ExoNewsDetailsApp.vue]:
```suggestion
      <h3>{{ $t('news.details.notFound') }}</h3>
```

File: webapp/src/main/webapp/news-details-app/components/ExoNewsDetailsApp.vue [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+  created() {
+    this.$newsServices.getNewsById(this.newsId, false)
+      .then(news => {
+        if (news !== 'error') {
+          this.news = news;
+          if (!this.news.spaceMember) {
+            this.$root.$emit('restricted-space', this.news.spaceDisplayName);
+          }
+        } else {
+          this.notFound = true;
```
Comment (azayati) [inline webapp/src/main/webapp/news-details-app/components/ExoNewsDetailsApp.vue:38]: can you test on news !== null here ? don't forget also that we should be in this case if the user is not allowed to see the news (not only news not existing)

File: webapp/src/main/webapp/news-details-app/components/ExoNewsDetailsApp.vue [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+        }
+      });
+  },
+  methods: {
+    redirectUrl() {
```
Comment (azayati) [inline webapp/src/main/webapp/news-details-app/components/ExoNewsDetailsApp.vue]: not needed

File: webapp/src/main/webapp/news-details-app/index.jsp [hunk 14]
```diff
@@ -0,0 +1,10 @@
+<div class="VuetifyApp">
+  <div id="NewsDetailApp">
+  </div>
+</div>
+
+<script>
+  require(['PORTLET/news/NewsDetail'], function(newsDetails) {
```
Comment (azayati) [inline webapp/src/main/webapp/news-details-app/index.jsp:7]:
```suggestion
  require(['PORTLET/news/NewsDetails'], function(newsDetails) {
```

File: webapp/src/main/webapp/news-details-app/index.jsp [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<div class="VuetifyApp">
+  <div id="NewsDetailApp">
```
Comment (azayati) [inline webapp/src/main/webapp/news-details-app/index.jsp:2]:
```suggestion
  <div id="NewsDetailsApp">
```

File: webapp/src/main/webapp/news-details-app/main.js [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+if (extensionRegistry) {
+  const components = extensionRegistry.loadComponents('NewsDetailApp');
+  if (components && components.length > 0) {
+    components.forEach(cmp => {
+      Vue.component(cmp.componentName, cmp.componentOptions);
+    });
+  }
+}
+
+let newsDetailApp;
```
Comment (azayati) [inline webapp/src/main/webapp/news-details-app/main.js]:
```suggestion
let newsDetailsApp;
```

use newsDetails everywhere

File: webapp/src/main/webapp/news-details-app/initComponents.js [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import * as  newsServices from '../services/newsServices';
+import ExoNewsDetailsApp from './components/ExoNewsDetailsApp.vue';
+const components = {
+  'exo-news-detail-app': ExoNewsDetailsApp,
```
Comment (azayati) [inline webapp/src/main/webapp/news-details-app/initComponents.js]:
```suggestion
  'exo-news-details-app': ExoNewsDetailsApp,
```

File: webapp/src/main/webapp/news-details-app/main.js [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+    // init Vue app when locale resources are ready
+    newsDetailApp = new Vue({
+      el: '#NewsDetailApp',
+      data: function() {
+        return {
+          newsId: getURLQueryParam('newsId'),
+        };
+      },
+      template: `<v-app id="${appId}">
+                   <exo-news-detail-app :news-id="newsId"></exo-news-detail-app>
```
Comment (azayati) [inline webapp/src/main/webapp/news-details-app/main.js]:
```suggestion
                   <exo-news-details-app :news-id="newsId"></exo-news-details-app>
```

File: webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue [hunk 19]
```diff
@@ -209,7 +219,7 @@ export default {
       return this.news && `${eXo.env.portal.context}/${eXo.env.portal.portalName}/profile/${this.news.updater}`;
     },
     notSameUpdater() {
-      return this.news && (this.news.updater !== this.news.author || this.news.authorFullName !== this.news.updaterFullName);
+      return this.news.updater !=='__system' && this.news && (this.news.updater !== this.news.author || this.news.authorFullName !== this.news.updaterFullName);
```
Comment (azayati) [inline webapp/src/main/webapp/news-details/components/ExoNewsDetails.vue]:
```suggestion
      return this.news && this.news.updater !=='__system' && (this.news.updater !== this.news.author || this.news.authorFullName !== this.news.updaterFullName);
```

File: webapp/webpack.common.js [hunk 20]
```diff
@@ -14,6 +14,7 @@ let config = {
     news :'./src/main/webapp/news/main.js',
     newsSearchCard: './src/main/webapp/news-search/main.js',
     newsExtensions: './src/main/webapp/news-extensions/main.js',
+    newsDetailApp: './src/main/webapp/news-details-app/main.js',
```
Comment (azayati) [inline webapp/webpack.common.js]:
```suggestion
    newsDetailsApp: './src/main/webapp/news-details-app/main.js',
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 21]
```diff
@@ -224,6 +227,9 @@ public Response getNews(@Context HttpServletRequest request,
       } else {
         news = newsService.getNews(newsFilter);
       }
+      if (news == null) {
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: to be deleted

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 22]
```diff
@@ -302,17 +302,19 @@ public Response getNewsById(@Context HttpServletRequest request,
...
         return Response.status(Response.Status.BAD_REQUEST).build();
       }
-
+      String authenticatedUser = request.getRemoteUser();
       News news = newsService.getNewsById(id, editMode);
       if (news == null) {
         return Response.status(Response.Status.NOT_FOUND).build();
       }
-
+      if (StringUtils.equals(news.getPublicationState(), PublicationDefaultStates.STAGED)
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java:310]: Add draft also here, check that a draft can be accessible by only author

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Review by azayati — DISMISSED** [review]
LGTM after QA validation

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 23]
```diff
@@ -173,6 +173,9 @@ public Response scheduleNews(@Context HttpServletRequest request,
       if (news == null) {
         return Response.status(Response.Status.NOT_FOUND).build();
       }
+      if (!Boolean.TRUE.equals(news.isCanEdit())) {
```
Comment (hbenali) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java:176]: Could we improve this ? cc @azayati 
```suggestion
      if (Boolean.FALSE.equals(news.isCanEdit())) {
```
or 
```suggestion
      if (!news.isCanEdit()) {
```
or 
```suggestion
      if (news.isCanEdit() == false) {
```

Comment (azayati) [reply services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java:176]: it was !news.isCanEdit(), but we have changed as recommended by Sonar Check
if (Boolean.FALSE.equals(news.isCanEdit())) { seems good for me

Comment (hbenali) [reply services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java:176]: cc @hassenelkamel-exo

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**